homeassistant/components/upnp/__init__.py

```diff
@@ -82,7 +83,7 @@ async def device_discovered(
     assert discovery_info is not None
     assert discovery_info.ssdp_udn
     assert discovery_info.ssdp_all_locations
-    force_poll = False
+    force_poll = entry.options.get(CONFIG_ENTRY_FORCE_POLL, False)
```

Comment:
Generally I don't like options because they increase complexity.

Given the wide range of devices this integration is supporting with various levels of quality, and different implementations of the spec, I think its unavoidable in this case.